src/Data/List/Fresh.agda

```diff
@@ -161,29 +161,28 @@ module _ {P : Pred A p} (P? : U.Decidable P) where
   takeWhile-# : ∀ {R : Rel A r} a (as : List# A R) → a # as → a # takeWhile as
 
   takeWhile []             = []
-  takeWhile (cons a as ps) with does (P? a)
-  ... | true  = cons a (takeWhile as) (takeWhile-# a as ps)
-  ... | false = []
+  takeWhile (cons a as ps) =
+    if does (P? a) then cons a (takeWhile as) (takeWhile-# a as ps) else []
 
+  -- this 'with' is crucial to get things to reduce enough
```

Comment:
Which 'things'? Better, I think, for the comment to better 'earn' its place in the code, would be to spell out that you want (at least! but what else?) `takeWhile (a ∷# as)` to reduce in the *type* of this branch?

Reply:
Agreed and fixed